unstructured/partition/api.py

```diff
@@ -97,6 +136,94 @@ def partition_via_api(
         )
 
 
+def get_retries_config(
+    retries_connection_errors: Optional[bool],
+    retries_exponent: Optional[float],
+    retries_initial_interval: Optional[int],
+    retries_max_elapsed_time: Optional[int],
+    retries_max_interval: Optional[int],
+    sdk: UnstructuredClient,
+) -> Optional[retries.RetryConfig]:
+    """Constructs a RetryConfig object from the provided parameters. If any of the parameters
+    are None, the default values are taken from the SDK configuration or the default constants.
+
+    If all parameters are None, returns None (and the SDK-managed defaults are used within the
+    client)
+
+    The solution is not perfect as the RetryConfig object does not include the defaults by
+    itself so we might need to construct it basing on our defaults.
+
+    Parameters
+    ----------
+    retries_connection_errors
+        Defines whether to retry on connection errors.
+    retries_exponent
+        Defines the exponential factor to increase the interval between retries.
+    retries_initial_interval
+        Defines the time interval to wait before the first retry in case of a request failure.
+    retries_max_elapsed_time
+        Defines the maximum time to wait for retries. If exceeded, the original exception is raised.
+    retries_max_interval
+        Defines the maximum time interval to wait between retries.
+    sdk
+        The UnstructuredClient object to take the default values from.
+    """
+    retries_config = None
+    default_retries_config = sdk.sdk_configuration.retry_config
+    if any(
+        setting is not None
+        for setting in (
+            retries_initial_interval,
+            retries_max_interval,
+            retries_exponent,
+            retries_max_elapsed_time,
+            retries_connection_errors,
+        )
+    ):
+        default_retries_initial_interval = (
+            default_retries_config.backoff.initial_interval
+            if default_retries_config and default_retries_config.backoff.initial_interval
+            else DEFAULT_RETRIES_INITIAL_INTERVAL_SEC
+        )
```

Comment:
small nit; this pattern repeated enough times for us to consider refactor like
```suggestion
        def _config_or_default(attr: str, default_val: int | float):
            return (getattr(default_retries_config.backoff, attr) if default_retries_config else None) or default_val
        default_retries_initial_interval = _config_or_default("initial_interval", DEFAULT_RETRIES_INITIAL_INTERVAL_SEC)
        # and those below
```

Reply:
and one more note this logic means if the config value is 0 we will be using default; is that intended or should we change the second condition to `default_retries_config.backoff.initial_interval is not None` (I got burned by this recently 😬 )

Reply:
I've started the implementation to check for `None` (as you suggested) but in this specific case all these settings should have values greater than 0 (zero is incorrect) - so this simplified check should be ok.
I'll mention it in the comments.